src/argilla/client/feedback.py

```diff
@@ -0,0 +1,454 @@
+            raise TypeError(f"Expected `fields` to be a list, got {type(fields)} instead.")
+        any_required = False
+        for field in fields:
+            if not isinstance(field, FieldSchema):
+                raise TypeError(f"Expected `fields` to be a list of `FieldSchema`, got {type(field)} instead.")
+            if not any_required and field.required:
+                any_required = True
+        if not any_required:
+            raise ValueError("At least one `FieldSchema` in `fields` must be required (`required=True`).")
+        self.__fields = fields
+        self.__fields_schema = None
+
+        if not isinstance(questions, list):
+            raise TypeError(f"Expected `questions` to be a list, got {type(questions)} instead.")
+        any_required = False
+        for question in questions:
+            if not isinstance(question, QuestionSchema):
+                raise TypeError(f"Expected `questions` to be a list of `QuestionSchema`, got {type(question)} instead.")
+            if not any_required and question.required:
+                any_required = True
+        if not any_required:
+            raise ValueError("At least one `QuestionSchema` in `questions` must be required (`required=True`).")
+        self.__questions = questions
+
+        self.__records = []
+
+    def __len__(self) -> int:
+        return len(self.__records)
+
+    def __getitem__(self, key: Union[slice, int]) -> Union["FeedbackItemModel", List["FeedbackItemModel"]]:
+        if len(self.__records) < 1:
+            raise RuntimeError(
+                "In order to get items from `rg.FeedbackDataset` you need to either add them first with `add_records` or fetch them from Argilla or HuggingFace with `fetch_records`."
+            )
+        if isinstance(key, int) and len(self.__records) < key:
+            raise IndexError(f"This dataset contains {len(self)} records, so index {key} is out of range.")
+        return self.__records[key]
+
+    def __del__(self) -> None:
+        if hasattr(self, "__records"):
+            del self.__records
+
+    def __enter__(self) -> "FeedbackDataset":
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        self.__del__()
+
+    @property
+    def guidelines(self) -> str:
+        return self.__guidelines
+
+    @guidelines.setter
+    def guidelines(self, guidelines: str) -> None:
+        if not isinstance(guidelines, str):
+            raise TypeError(f"Expected `guidelines` to be a string, got {type(guidelines)} instead.")
+        self.__guidelines = guidelines
+
+    @property
+    def fields(self) -> List[FeedbackFieldModel]:
+        return self.__fields
+
+    @property
+    def questions(self) -> List[FeedbackQuestionModel]:
+        return self.__questions
+
+    @property
+    def records(self) -> List["FeedbackItemModel"]:
+        return self.__records
+
+    def fetch_records(self, overwrite: bool = False) -> List["FeedbackItemModel"]:
```

Comment:
So users can refresh info from argilla using this method. How users can fetch a chunk of data `ds.records[100:1000]` ?

Reply:
The thing is that the only consistent solution is to fetch all the records from Argilla when calling `from_argilla`, as otherwise relying on the API for every single record will end up on a consistency issue... We can discuss this later today with a demo!